src/lib/engine/index.ts

```diff
@@ -61,6 +66,69 @@ export function runGame(code: string, canvas: HTMLCanvasElement, onPageError: (e
 			const playTuneRes = playTune(tune, n)
 			tunes.push(playTuneRes)
 			return playTuneRes
+		},
+		console: {
+			...console,
+			log: (...args: any[]) => {
+				console.log(...args)
+				logInfo.value = [...logInfo.value, {
+					args: args,
+					nums: (()=>{
+						var err = getErrorObject();
+						// get <anonymous>:x:x if on chrome, Function:x:x on firefox
+						// @ts-ignore
+						var stack = err.stack
+						// check browser
+						var browser = navigator.userAgent;
+						if(browser.indexOf("Chrome") != -1) {
```

Comment:
Suggest breaking this out into several smaller functions and cleaning it up for readability.  I.e. here just have a:

`return extractConsoleErrorNums(...)`

and within that (this is all pseudocode):

```
function extractConsoleErrorNums(...) {
  const browser = getBrowser(userAgent)
  switch(browser) {
    case CHROME: return extractConsoleErrorNumsForChrome(...); break;
    default: return extractConsoleErrorNumsDefault(...); break;
  }
}
```

Reply:
i meant to do that and forgot, will do